flopy/mf6/utils/lakpak_utils.py

```diff
@@ -0,0 +1,219 @@
+            embedded = False
+
+    connection_dict = {}
+    connectiondata = []
+
+    # find unique lake numbers
+    unique = np.unique(lake_map)
+
+    # exclude lakes with lake numbers less than 0
+    idx = np.where(unique > -1)
+    unique = unique[idx]
+
+    # get the model grid elevations
+    if embedded:
+        elevations = modelgrid.top_botm
+    else:
+        elevations = None
+
+    dx, dy = None, None
+
+    # embedded lakes
+    for lake_number in unique:
+        iconn = 0
+        indices = np.argwhere(lake_map == lake_number)
+        for index in indices:
+            cell_index = tuple(index.tolist())
+            if embedded:
+                leak_value = bedleak[cell_index[1:]]
+                if modelgrid.grid_type == "structured":
+                    if dx is None:
+                        xv, yv = modelgrid.xvertices, modelgrid.yvertices
+                        dx = xv[0, 1:] - xv[0, :-1]
+                        dy = yv[:-1, 0] - yv[1:, 0]
+                    (
+                        cellids,
+                        claktypes,
+                        belevs,
+                        televs,
+                        connlens,
+                        connwidths,
+                    ) = __structured_lake_connections(
+                        lake_map, cell_index, dx, dy, elevations
+                    )
+                elif modelgrid.grid_type == "vertex":
+                    raise NotImplementedError(
+                        "embedded lakes have not been implemented"
+                    )
+            else:
+                leak_value = bedleak[cell_index]
+                cellids = [(0,) + cell_index]
+                claktypes = ["vertical"]
+                belevs = [0.0]
+                televs = [0.0]
+                connlens = [0.0]
+                connwidths = [0.0]
+
+            # iterate through each cellid
+            for (cellid, claktype, belev, telev, connlen, connwidth) in zip(
+                cellids, claktypes, belevs, televs, connlens, connwidths
+            ):
+                connectiondata.append(
+                    [
+                        lake_number,
+                        iconn,
+                        cellid[:],
+                        claktype,
+                        leak_value,
+                        belev,
+                        telev,
+                        connlen,
+                        connwidth,
+                    ]
+                )
+                iconn += 1
+
+        # set number of connections for lake
+        connection_dict[lake_number] = iconn
+    return connection_dict, connectiondata
+
+
+def __structured_lake_connections(lake_map, cell_index, dx, dy, elevations):
+    nlay, nrow, ncol = lake_map.shape
+    cellids = []
+    claktypes = []
+    belevs = []
+    televs = []
+    connlens = []
+    connwidths = []
+
+    k, i, j = cell_index
+
+    # back face
+    if i > 0:
+        ci = (k, i - 1, j)
+        if lake_map[ci] < 0:
```

Comment:
Lake cells have to be zero and greater. Non-lake cells can be any value less than 0. 

Not sure if `np.nan` would work. Should it?

Any thoughts on a better way to pass in lake identifiers?

Reply:
Could used a masked array; np.nan would also work.  Also, could pass in IDOMAIN and also pass it back if incised lakes should change it.

Reply:
Masked array could work. Should decide and do one or the other not all options.

Seems like the user should handle the `idomain` modification outside of the function since they would still have to manually update idomain if loading an existing model.

Reply:
Implemented lake_map evaluations as masked arrays. What do you think?

Reply:
Yeah, I like that.  Looks good.